pkg/apis/core/validation/validation.go

```diff
@@ -354,6 +354,13 @@ func ValidateVolumes(volumes []core.Volume, podMeta *metav1.ObjectMeta, fldPath
 	allErrs := field.ErrorList{}
 
 	allNames := sets.String{}
+	allCreatedPVCs := sets.String{}
+	for _, vol := range volumes {
+		// Remember which PVCs will be created for this pod.
+		if podMeta != nil && vol.VolumeSource.Ephemeral != nil {
+			allCreatedPVCs.Insert(podMeta.Name + "-" + vol.Name)
```

Comment:
So this doesn't take into account generated name?  Shouldn't this skip this chunk if pod GenerateName is not empty?

Reply:
As I learned now, podMeta.Name *is* always set already when validating a pod:

https://github.com/kubernetes/kubernetes/blob/896da2253c77d15fc551843e515039c6e9c86c99/pkg/apis/core/validation/validation.go#L3116-L3120
https://github.com/kubernetes/kubernetes/blob/896da2253c77d15fc551843e515039c6e9c86c99/pkg/apis/core/validation/validation.go#L328-L339
https://github.com/kubernetes/apimachinery/blob/fb3c024b9997b3ee34b110f9c47216b7e7bf3a40/pkg/api/validation/objectmeta.go#L154-L155

I changed the code such that it skips the check if podMeta is nil (non-Pod validation) or podMeta.Name is empty (an error that gets flagged by `ValidateObjectMeta`).